instrumentation/jdbi3/src/main/java/brave/jdbi3/Jdbi3BravePlugin.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package brave.jdbi3;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.spi.JdbiPlugin;
+import org.jdbi.v3.core.statement.SqlStatements;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.StatementContextListener;
+
+import brave.Span;
+import brave.Tracing;
+
+/**
+ * Instrumentation for JDBI 3.
+ * <p>
+ * JDBI plugin that sends a trace span for every SQL command executed by Jdbi.
+ * Install by calling {@code jdbi.installPlugin(new Jdbi3BravePlugin(tracing));}
+ */
+public class Jdbi3BravePlugin extends JdbiPlugin.Singleton {
+    private final Tracing tracing;
+    private final boolean includeBindVariables;
+    private static final Pattern serviceNamePattern = Pattern.compile("(^|&)zipkinServiceName=(?<serviceName>[^&]*)");
+
+    private Jdbi3BravePlugin(Tracing tracing, boolean includeBindVariables) {
+        this.tracing = tracing;
+        this.includeBindVariables = includeBindVariables;
+    }
+
+    public static Builder newBuilder(Tracing tracing) {
+        return new Builder(tracing);
+    }
+
+    @Override
+    public void customizeJdbi(Jdbi jdbi) {
+        if (tracing == null) {
```

Comment:
remove tracing == null here to reduce branching, as it is a bug to have null here and we can guard against it.